src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const arrayResult = [];
+  let result = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      result = Object.assign(result, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete result[key];
+      }
+    }
 
+    if (action.type === 'clear') {
+      for (const clear in result) {
+        delete result[clear];
+      }
+    }
```

Comment:
[CODE STYLE]: switch/case should always have a default case for error handling. Even though you are not using a switch statement, it's important to note that if you were, a default case should be included to handle any unexpected action types.